### PR TITLE
[knx] Bugfix problem with DPT 251.600

### PR DESCRIPTION
**Pull request by genesis81**
In the old implementation the datatype was implemented with byte 0-255. 
The DPT 251.600 is defined with 0-100%, change this from byte to PercentType.

Signed-off-by: Marco Müller <marco@ms-mueller.ch>

### CODE REVIEW

**Comment by J-N-K** [comment]
Technically this is correct. The question is what actual problem occurs with the current implementation, because this change is breaking and I can't see any real issue with using 0-255.

**Comment by genesis81** [comment]
When i try to send a new color to the bus i have always an error out of range. 

See my detailed description in this threat: #https://community.openhab.org/t/rgbw-color-out-of-range-dpt-251-600/150055

**Comment by genesis81** [comment]
> Technically this is correct. The question is what actual problem occurs with the current implementation, because this change is breaking and I can't see any real issue with using 0-255.

I have checked the code again. I don't see that the change is breaking, actual we have a bug. When you send a value over 100 calimero reject the request. That means all value over 100 is not working and will not send to the knx bus.

**Comment by J-N-K** [comment]
Correct, I misread the change. I would suggest a minor change though, to get higher accuracy.

**Comment by genesis81** [comment]
> Correct, I misread the change. I would suggest a minor change though, to get higher accuracy.

I'm complete open for any suggest, please tell me what we can optimize.

**Comment by holgerfriedrich** [comment]
@J-N-K @genesis81 Give me some time, just trying to get a proper test which includes the Calimero part. Until now we have only tests for ValueEncoder decode/encode. With a custom KNXNetworkLink implementation, we should see everything down to the KNX frame level including the coding done in Calimero.

File: bundles/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/internal/dpt/ValueEncoder.java [hunk 1]
```diff
@@ -162,8 +162,8 @@ private static String handleHSBType(String dptId, HSBType hsb) {
                 double[] xyY = ColorUtil.hsbToXY(hsb);
                 return String.format("(%,.4f %,.4f) %,.1f %%", xyY[0], xyY[1], xyY[2] * 100.0);
             case "251.600":
-                rgb = ColorUtil.hsbToRgb(hsb);
-                return String.format("%d %d %d - %%", rgb[0], rgb[1], rgb[2]);
+                PercentType[] rgbw = ColorUtil.hsbToRgbPercent(hsb);
+                return String.format("%d %d %d - %%", rgbw[0].intValue(), rgbw[1].intValue(), rgbw[2].intValue());
```
Comment (J-N-K) [inline bundles/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/internal/dpt/ValueEncoder.java:166]:
```suggestion
                return String.format("%,.1f %,.1f %,.1f - %%", rgbw[0].doubleValue(), rgbw[1].doubleValue(), rgbw[2].doubleValue());
```

**Review by J-N-K — CHANGES_REQUESTED** [review]

**Comment by J-N-K** [comment]
@holgerfriedrich Really nice. I think this can be fixed anyway, I check in calimero code that a float in the range from 0 to 100 is expected here.

**Review by J-N-K — APPROVED** [review]
LGTM, @holgerfriedrich do you agree?

**Comment by J-N-K** [comment]
I think you need to apply spotless

**Comment by J-N-K** [comment]
Test failures are independent, I also see them in my local builds without any changes.

**Review by J-N-K — APPROVED** [review]

**Comment by genesis81** [comment]
> Test failures are independent, I also see them in my local builds without any changes.

Looks like changes in the core system that break the tests.

**Comment by J-N-K** [comment]
I already commented in the PR that includes the cause of these problems.

**Review by holgerfriedrich — APPROVED** [review]
LGTM, thanks for providing this bugfix!

Just for reference: openhab/openhab-core#3834 (core change currently breaking CI test) and #15727 (WIP new tests)

**Pull request merged**